nimare/annotate/gclda.py

```diff
@@ -1,24 +1,247 @@
 """Topic modeling with generalized correspondence latent Dirichlet allocation."""
+import itertools
 import logging
+import math
 import os.path as op
 
 import nibabel as nib
 import numpy as np
 import pandas as pd
+from joblib import Parallel, delayed
 from nilearn._utils import load_niimg
+from numba import jit
 from scipy.stats import multivariate_normal
 
 from nimare.base import NiMAREBase
-from nimare.utils import get_template
+from nimare.utils import _check_ncores, get_template
 
 LGR = logging.getLogger(__name__)
 
 
+@jit(nopython=True)
+def _update_word_topic_assignments(
+    word_topic_idx,
+    nz_word_topic,
+    nz_sum_topic,
+    nz_doc_topic,
+    ny_doc_topic,
+    word_doc_idx,
+    word_idx,
+    voc_len,
+    beta,
+    gamma,
+    randseed,
+):
+    """Update wtoken_topic_idx (z) indicator variables assigning words->topics.
+
+    Parameters
+    ----------
+    word_topic_idx : :obj:`numpy.ndarray`
+        Topic assignment for word tokens.
+    nz_word_topic : (W x T) :obj:`numpy.ndarray`
+        Number of word-tokens assigned to each topic (T) per word-type (W).
+    nz_sum_topic : (1 x T) :obj:`numpy.ndarray`
+        Total number of word-tokens assigned to each topic (T) (across all docs).
+    nz_doc_topic : (D x T) :obj:`numpy.ndarray`
+        Number of word-tokens assigned to each topic (T) per document (D).
+    ny_doc_topic : (D x T) :obj:`numpy.ndarray`
+        Number of peak-tokens assigned to each topic (T) per document (D).
+    word_doc_idx : :obj:`numpy.ndarray`
+        Document index.
+    word_idx : :obj:`numpy.ndarray`
+        Word tokens.
+    voc_len : :obj:`int`
+        Total number of word-tokens in the vocabulary.
+    beta : :obj:`float`
+        Prior count on word-types for each topic.
+    gamma : :obj:`float`
+        Prior count added to y-counts when sampling z assignments.
+    randseed : :obj:`int`
+        Random seed for this iteration.
+
+    Returns
+    -------
+    word_topic_idx : :obj:`numpy.ndarray`
+        Updated topic assignment for word tokens.
+    nz_word_topic : (W x T) :obj:`numpy.ndarray`
+        Updated number of word-tokens assigned to each topic (T) per word-type (W).
+    nz_sum_topic : (1 x T) :obj:`numpy.ndarray`
+        Updated total number of word-tokens assigned to each topic (T) (across all docs).
+    nz_doc_topic : (D x T) :obj:`numpy.ndarray`
+        Updated number of word-tokens assigned to each topic (T) per document (D).
+    """
+    # --- Seed random number generator
+    np.random.seed(randseed)
```

Comment:
It's likely jit does something slightly different here, since the results change if I comment out `@jit` above.